src/main/kotlin/ar/edu/unahur/obj2/tareas/Tarea.kt

```diff
@@ -1,3 +1,67 @@
 package ar.edu.unahur.obj2.tareas
 
-class Tarea
+interface ITarea {
+    fun horasNecesariasParaFinalizarTarea(): Int
+    fun costoDeUnaTarea(): Double
+    fun calcularSalarioTotalDeEmpleados(): Double
+    fun darNominaDeEmpleados(): MutableList<Empleado>
+}
+
+class TareaSimple(
+    val cantidadHorasEstimadas: Int,
+    val responsable: Empleado,
+    var empleados: MutableList<Empleado>,
+    val costoInfraestructura: Int
+) : ITarea {
+    override fun horasNecesariasParaFinalizarTarea() = cantidadHorasEstimadas / empleados.size
+
+    override fun costoDeUnaTarea(): Double {
+        return costoInfraestructura + this.calcularSalarioTotalDeEmpleados() + responsable.calcularCuantoCobraPorCantidadDeHorasTrabajadas(
+            cantidadHorasEstimadas
+        )
+    }
+
+    override fun calcularSalarioTotalDeEmpleados(): Double {
+        return empleados.sumByDouble { it.calcularCuantoCobraPorCantidadDeHorasTrabajadas(this.horasNecesariasParaFinalizarTarea()) }
+    }
+
+    override fun darNominaDeEmpleados(): MutableList<Empleado> {
+        var nominaEmpleados = empleados
+        nominaEmpleados.add(responsable)
+        return nominaEmpleados
+    }
+
+}
+
+class TareaIntegracion(val responsable: Empleado) : ITarea {
+
+    var listaDeTareas = mutableListOf<ITarea>()
+
+    override fun horasNecesariasParaFinalizarTarea(): Int {
+        return listaDeTareas.sumBy { it.horasNecesariasParaFinalizarTarea() } + (listaDeTareas.sumBy { it.horasNecesariasParaFinalizarTarea() } / 8)
+    }
+
+    override fun costoDeUnaTarea() = costoSubTareas() + bonus()
+
+    fun costoSubTareas() = listaDeTareas.sumByDouble { it.costoDeUnaTarea() }
+    fun bonus() = costoSubTareas() * 0.03
+
+    override fun calcularSalarioTotalDeEmpleados(): Double {
+        return listaDeTareas.sumByDouble { it.calcularSalarioTotalDeEmpleados() }
+    }
+
+    override fun darNominaDeEmpleados(): MutableList<Empleado> {
+        var listaCompleta = listaDeTareas.flatMap { it.darNominaDeEmpleados() }.toMutableList()
+        listaCompleta.add(responsable)
+        return listaCompleta
```

Comment:
Acá también se podía resolver sin usar variables:

```suggestion
        return listaDeTareas.flatMap { it.darNominaDeEmpleados() } + responsable
```